File: src/main/java/com/amm/kafkafirst/interfaces/ProviderController.java
package com.amm.kafkafirst.interfaces;

import com.amm.kafkafirst.application.ProducerService;
import com.amm.kafkafirst.infrastructure.Message;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;

import java.time.LocalDate;

@Controller
public class ProviderController {

    private ProducerService producerService;

    public ProviderController(@Autowired ProducerService producerService){
        this.producerService = producerService;
    }

    @GetMapping("/provider")
    public ResponseEntity<HttpStatus> index() {
        final Message message = new Message(200, "Hello World " + LocalDate.now());
        producerService.sendMessage(message);
        return new ResponseEntity<>(HttpStatus.OK);
    }
}

File: src/main/kotlin/com/amm/kafkafirst/infrastructure/interfaces/ProviderController.kt
package com.amm.kafkafirst.infrastructure.interfaces

import com.amm.kafkafirst.infrastructure.services.ProducerService
import com.amm.kafkafirst.infrastructure.kafka.Message
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping

import java.time.LocalDate
import org.springframework.web.bind.annotation.RestController

@RestController
class ProviderController(@param:Autowired private val producerService: ProducerService) {

    @GetMapping("/provider")
    fun index(): ResponseEntity<HttpStatus> {
        val message = Message(200, "Hello World " + LocalDate.now())
        producerService.sendMessage(message)
        return ResponseEntity(HttpStatus.OK)
    }
}